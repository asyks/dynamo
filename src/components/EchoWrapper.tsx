import React from 'react'

import AttentedWebSocket, { messageEventWrapper } from '../modules/WebSocket'

import { wsConnectHandler, wsSendHandler, wsDisconnectHandler } from './types'
import EchoForm from './EchoForm'

const EchoWrapper: React.FC = () => {
  const wsUri = "wss://echo.websocket.org/"
  let websocket: WebSocket

  const connect: wsConnectHandler = (eventCallback) => {
    websocket = new AttentedWebSocket(
      wsUri,
      evt => {
        messageEventWrapper(evt, eventCallback)
      },
    )
  }

  const disconnect: wsDisconnectHandler = () => {
    websocket.close()
  }

  const send: wsSendHandler = (message, eventCallback) => {
    const messagePrefixed = "SENT: " + message
    websocket.send(message)
    eventCallback(messagePrefixed)
  }

  return (
    <EchoForm
      connect={connect}
      disconnect={disconnect}
      send={send}
    />
  )
}

export default EchoWrapper
