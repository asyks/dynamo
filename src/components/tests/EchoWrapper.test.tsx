import React from 'react'

import renderer from 'react-test-renderer'

import EchoWrapper from '../EchoWrapper'

describe("components/EchoWrapper", () => {
  it("nominally renders", () => {
    expect(renderer.create(<EchoWrapper />).toJSON()).toMatchSnapshot()
  })
})
