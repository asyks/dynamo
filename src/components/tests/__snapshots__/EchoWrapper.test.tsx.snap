// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EchoWrapper nominally renders 1`] = `
<div>
  <div>
    websocket echo
  </div>
  <button
    onClick={[Function]}
  >
    Connect
  </button>
  <input
    onChange={[Function]}
    type="text"
    value="HelloWebSocket!"
  />
  <button
    onClick={[Function]}
  >
    Send
  </button>
  <button
    onClick={[Function]}
  >
    Disconnect
  </button>
  <ul
    className="Web-socket-echo-log"
  />
</div>
`;
